internal/service/advancedcluster/model_advanced_cluster.go

```diff
@@ -300,12 +300,9 @@ func UpgradeRefreshFunc(ctx context.Context, name, projectID string, client Clus
 	}
 }
 
-func ResourceClusterListAdvancedRefreshFunc(ctx context.Context, projectID string, client ClusterService) retry.StateRefreshFunc {
+func ResourceClusterListAdvancedRefreshFunc(ctx context.Context, projectID string, client admin.ClustersApi) retry.StateRefreshFunc {
 	return func() (any, string, error) {
-		params := &admin.ListClustersApiParams{
-			GroupId: projectID,
-		}
-		clusters, resp, err := client.List(ctx, params)
+		clusters, resp, err := client.ListClusters(ctx, projectID).Execute()
```

Comment:
why are we changing the function call, what is the difference?

Reply:
```golang
func (a *ClusterServiceFromClient) List(ctx context.Context, options *admin.ListClustersApiParams) (*admin.PaginatedAdvancedClusterDescription, *http.Response, error) {
	return a.client.ClustersApi.ListClustersWithParams(ctx, options).Execute()
```
👆 This was the old code in the `service_advanced_cluster.go`. (only named `List`)

I preferred using `ListClusters` instead of `ListClustersWithParams` since the `ListClustersApiParams` only used `GroupId` in the `admin.ListClustersApiParams`